docs/contribute/general/pr_workflow.rst

```diff
@@ -0,0 +1,128 @@
+- Include a clear and concise docstring which is formatted according to the project's documentation guidelines and make sure that it is added to the API documentation.
+- Keep contributions made in a PR as compact and module as possible. It is always beneficial to re-use existing functionality when suitable.
+- Please refrain from making changes on code sections that are not related to the PR. This is to ensure that the git history remains clear and improves maintainability and most importantly makes reviewing a PR easier.
+- Please avoid purely cosmetic formatting changes. Maintainers will usually ask you to revert unrelated or purely cosmetic changes for maintainability reasons.
+- Please keep the git history as clean as possible and write descriptive and meaningful commit messages. Commit messages such as `"add fix"` are not descriptive enough.
+
+Once all criteria are met, you can create the PR through the GitHub interface.
+
+When Creating the Pull Request
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Please make sure to consider the following points when creating the PR:
+
+- Provide a clear and descriptive title for the PR.
+- Write a detailed description of the changes made in the PR, including the motivation behind them and any relevant context.
+- If the PR addresses any existing issues, link them using the ``closes #issue-number`` syntax. Note that an existing issue does not mean that you can simply skip the PR description.
+- Browse the available labels and assign the most appropriate ones to your PR.
+- If possible, add the PR to the *Weekly meeting* project board. Workflow automation will then assign the status :backlog:`Backlog` which is automatically updated during the process.
+- If a milestone is already defined for the next release you can also assign it to the PR. Otherwise, maintainers will take care of this later. Please ignore the corresponding (failing) workflow check.
+
+
+The Review Process
+~~~~~~~~~~~~~~~~~~
+
+- If you open a PR which is not yet ready for review, mark it as "Draft". You can request reviews for draft PRs, but reviewers will know that it is not yet ready for final review and consequently provide conceptual feedback only. You can also indicate that you want to discuss the PR further by changing the status to :open-discussion:`Open Discussion` in the project section.
+- As soon as the PR is ready for review, request a review from the maintainer team of the respective package or project. The teams are named after the respective repository. Remember to change the status from "Draft" to "Ready for Review" if applicable. Please note that it might take some time for reviewers to get to your PR depending on their availability. If you have not received feedback after a reasonable amount of time, feel free to ping the maintainers once more or get in touch through slack or join a weekly developer meeting.
```

Comment:
I would consider this as part of preparing the PR, so I would suggest moving these two points to the list before.